### PR TITLE
refactor: Use pointer for connection receiver

### PR DESCRIPTION
**Pull request by erezrokah**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

Another lint related issue. I don't think it's a "real" issue as we update the DSN outside of `BuildFromConnParams`.
However it might be a bit confusing that the DNS is updated and the port and type aren't.

A smaller fix would be to only change `c Connection` to `c *Connection`

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/.github/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run --new-from-rev main` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -121,10 +121,14 @@ func (c Connection) BuildFromConnParams() (*dburl.URL, error) {
 	}
 	u.RawQuery = v.Encode()
 
-	return &dburl.URL{
+	url := &dburl.URL{
```
Comment (disq) [inline pkg/config/config.go:124]: bad variable name.

Comment (erezrokah) [reply pkg/config/config.go]: Yep :) Just realized it as the new linter errored on it

File: pkg/config/config.go [hunk 2]
```diff
@@ -121,10 +121,14 @@ func (c Connection) BuildFromConnParams() (*dburl.URL, error) {
 	}
 	u.RawQuery = v.Encode()
 
-	return &dburl.URL{
+	dbUrl := &dburl.URL{
```
Comment (disq) [inline pkg/config/config.go:124]: just do `c.DSN = &dburl.URL{params}.String()`?

Comment (erezrokah) [reply pkg/config/config.go]: Sure https://github.com/cloudquery/cloudquery/pull/782/commits/7138a8767724f70d3f8b024928807a837de0fb3d

**Review by roneli — APPROVED** [review]

**Pull request merged**